flocker/acceptance/endtoend/test_dataset.py

```diff
@@ -35,6 +40,112 @@ def test_dataset_creation(self, cluster):
         """
         return create_dataset(self, cluster)
 
+    def _get_package_source(self, default_version=None):
+        """
+        Get the package source for the flocker version under test from
+        environment variables.
+
+        :param unicode default_version: The version of flocker to use
+            if not specified in environment variables.
+
+        :return: A ``PackageSource`` that can be used to install the version of
+            flocker under test.
+        """
+        env_vars = ['FLOCKER_ACCEPTANCE_PACKAGE_BRANCH',
+                    'FLOCKER_ACCEPTANCE_PACKAGE_VERSION',
+                    'FLOCKER_ACCEPTANCE_PACKAGE_BUILD_SERVER']
+        missing_vars = list(x for x in env_vars if x not in os.environ)
+        if missing_vars:
+            raise SkipTest(
+                'Missing environment variables for upgrade test: %s.' %
+                ', '.join(missing_vars))
+        version = (os.environ['FLOCKER_ACCEPTANCE_PACKAGE_VERSION'] or
+                   default_version)
```

Comment:
Yeah that's my point. The environment variable should either be set or unset. It should never be an empty string.